### ISSUE TITLE
onInput.debounce(300 milliseconds).value does not trigger

### ISSUE THREAD

**Issue by fdietze** [OutWatch/outwatch#295]
```scala
import concurrent.duration._

input(
	tpe := "text",
	onInput.debounce(300 milliseconds).value --> ... // does not trigger
	onInput.value.debounce(300 milliseconds) --> ... // works correctly
)
```

**Comment by cornerman** [comment]
I have looked into this and the first actually throws an error:
```scala
scala.scalajs.js.JavaScriptException: TypeError: Cannot read property 'value' of null
```

Because things are mutable in dom world, we cannot really assure that you can access these things async after some time. We can at least only allow `value` and stuff on `EmitterBuilder.Sync`, but nothing is stopping you from doing `.map(_.value)` yourself :/

**Comment by cornerman** [comment]
And we should produce a proper error when people use `foreach`, because there is no `onError` method.

**Comment by cornerman** [comment]
Fixed in #355
